daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -1840,6 +1840,14 @@ func (devices *DeviceSet) initDevmapper(doInit bool) error {
 		if err := devicemapper.CreatePool(devices.getPoolName(), dataFile, metadataFile, devices.thinpBlockSize); err != nil {
 			return err
 		}
+		defer func() {
+			if retErr != nil {
```

Comment:
This works. The `return err` will assign the value to `retErr` https://tour.golang.org/welcome/1